kernel/aster-nix/src/net/socket/unix/stream/socket.rs

```diff
@@ -15,26 +15,30 @@ use crate::{
     },
     net::socket::{
         unix::{addr::UnixSocketAddrBound, UnixSocketAddr},
-        util::{send_recv_flags::SendRecvFlags, socket_addr::SocketAddr},
+        util::{
+            copy_packet_from_user, copy_packet_to_user, create_packet_buffer,
+            send_recv_flags::SendRecvFlags, socket_addr::SocketAddr, MessageHeader,
+        },
         SockShutdownCmd, Socket,
     },
     prelude::*,
     process::signal::Poller,
+    util::IoVec,
 };
 
-pub struct UnixStreamSocket(RwLock<State>);
+pub struct UnixStreamSocket(RwMutex<State>);
 
 impl UnixStreamSocket {
     pub(super) fn new_init(init: Init) -> Self {
-        Self(RwLock::new(State::Init(Arc::new(init))))
+        Self(RwMutex::new(State::Init(Arc::new(init))))
     }
 
     pub(super) fn new_listen(listen: Listener) -> Self {
-        Self(RwLock::new(State::Listen(Arc::new(listen))))
+        Self(RwMutex::new(State::Listen(Arc::new(listen))))
     }
 
     pub(super) fn new_connected(connected: Connected) -> Self {
-        Self(RwLock::new(State::Connected(Arc::new(connected))))
+        Self(RwMutex::new(State::Connected(Arc::new(connected))))
```

Comment:
Well, it seems that the Unix socket implementation is a copy of the early TCP/UDP implementation, and has diverged greatly as TCP/UDP sockets have been heavily refactored.

I'm not familiar with this, so is this change desired and what's the reason for it?

Reply:
It is undesired and is removed from this PR